solver/src/settlement.rs

```diff
@@ -59,22 +60,105 @@ impl Trade {
     }
 }
 
-pub trait Interaction: std::fmt::Debug + Send {
+#[cfg_attr(test, mockall::automock)]
+pub trait Interaction: Debug + Send {
     // TODO: not sure if this should return a result.
-    // Write::write returns a result but we know we write to a vector in memory so we know it will
-    // never fail. Then the question becomes whether interactions should be allowed to fail encoding
-    // for other reasons.
     fn encode(&self) -> Vec<EncodedInteraction>;
 }
 
+#[cfg(test)]
+impl Debug for MockInteraction {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        f.write_str("MockInteraction")
+    }
+}
+
+pub trait UnwrapInteraction: Interaction {
```

Comment:
Is this abstraction really useful if each solver needs to know the specific Interaction type a specific type of liquidity returns? Maybe in this case the LimtiOrder type should return the concrete UnwrapWethInteraction. How do you see it be reused?

Reply:
Yes, maybe it makes sense for limit orders to not return a boxed trait but a concrete type instead.